c_api.h

```diff
@@ -13,6 +13,8 @@
 #define MAX_ENTITY_NAME_LENGTH 32
 #define MAX_PURPOSE_LENGTH 64
 #define NETWORK_PROTOCOL_NAME_LENGTH 4
+#define MAX_SECURE_COMM_LENGTH \
+    1091  // Total length of message including header and encryption.
```

Comment:
1. Where is `MAX_PAYLOAD_LENGTH` defined? Should it be defined here as well?
2. I think the naming `MAX_SECURE_COMM_LENGTH` sounds a bit awkward. 
3. How about, we keep `MAX_PAYLOAD_LENGTH` as 1024, and define `MAX_SECURE_COMM_MSG_LENGTH` as something like this?:
`#define MAX_SECURE_COMM_MSG_LENGTH (MSG_TYPE_SIZE + 2 * SESSION_KEY_ID_SIZE + MAX_PAYLOAD_LENGTH)
-> This way, the code will be much more readable (e.g., we don't have to wonder why do we have this 1091 weird number?) and we don't have to change all `#define`'s everytime we change other constants, for example, what are you going to do fi you change `SESSION_KEY_ID_SIZE` from `8` to `12` or even to `16`? Are you going to manually compute all changes everytime?